docs/design/quantity_kind.rst

```diff
@@ -0,0 +1,52 @@
+.. namespace:: units
+
+quantity_kind
+=============
+
+The ``quantity_kind`` class template provides a similar interface to ``quantity``.
+
+Kinds
+-----
+
+The first template parameter of ``quantity_kind`` is a ``Kind``.
+
+``Kind``s, by themselves:
+
+    * Wrap a ``Dimension``, and
+    * opt into the downcasting facility.
+
+If a dimensional analysis operator on a quantity of kind *A*
+can result on a quantity of kind *B*,
+*A* and *B* are *related*.
+
+The library provides two ``Kind`` bases, ``kind`` and ``derived_kind``.
+
+``kind`` is an entry point, and there's only one per set of related kinds.
+
+``derived_kind`` is used to explicitly name other related kinds.
```

Comment:
Shouldn't we replace the order or arguments for `derived_kind` to be consistent with a `deduced_unit` (dim, kind)?

Reply:
Also, shouldn't we replace the order or arguments for `derived_kind` to be consistent with a `deduced_unit` (dim, kind)?

Reply:
Definitely.

Reply:
Opened #229 for that.